controlplane/kubeadm/internal/controllers/remediation_test.go

```diff
@@ -42,6 +42,50 @@ import (
 )
 
 func TestGetMachineToBeRemediated(t *testing.T) {
+	t.Run("returns the oldest of the provisioning machines with RemediateMachineAnnotation", func(t *testing.T) {
```

Comment:
can we change this test to ensure we alway pick the oldest machine with remediate annotation, no matter if there are provisioning machines or not?